### PR TITLE
Flake fixed

### PR DESCRIPTION
**Pull request by max-sixty**
Towards https://github.com/pydata/xarray/issues/2627

### CODE REVIEW

File: xarray/backends/netcdf3.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 # Special characters that are permitted in netCDF names except in the
 # 0th position of the string
-_specialchars = '_.@+- !"#$%&\()*,:;<=>?[]^`{|}~'
+_specialchars = '_.@+- !"#$%&\\()*,:;<=>?[]^`{|}~'
```
Comment (shoyer) [inline xarray/backends/netcdf3.py]: Good catch!

File: xarray/core/dataset.py [hunk 2]
```diff
@@ -2905,7 +2906,7 @@ def reduce(self, func, dim=None, keep_attrs=None, numeric_only=False,
         else:
             dims = set(dim)
 
-        missing_dimensions = [dim for dim in dims if dim not in self.dims]
+        missing_dimensions = [dim for d in dims if d not in self.dims]
```
Comment (shoyer) [inline xarray/core/dataset.py]: Here and below: This looks like a mistake -- `dim` is no longer the loop variable.

Comment (max-sixty) [reply xarray/core/dataset.py]: Thanks!

**Comment by max-sixty** [comment]
Let's merge this to ensure we don't have merge errors, and then answer https://github.com/pydata/xarray/issues/2627#issuecomment-449684619 after?

**Pull request merged**